usersimcrs/items/item_collection.py

```diff
@@ -122,3 +123,23 @@ def get_possible_property_values(self, property: str) -> Set[Any]:
                 else:
                     values.add(value)
         return values
+
+    def get_items_by_properties(
+        self, annotations: List[Annotation]
+    ) -> List[Item]:
+        """Returns a list of items that match the given utterance annotations.
+
+        Args:
+            annotations: List of annotation.
+
+        Returns:
+            List of matching items.
+        """
+        matching_items: List[Item] = []
+        for item in self._items.values():
```

Comment:
Looping through all the items for each lookup is very inefficient. It's fine for now, but let's create a new issue for this. We should either index items by properties or make some assumptions on how this will be used. E.g., if this is only used when the user is recommended an item, can we assume that items will be referred to by name?

Reply:
That's a good point. For now the items are indexed by ID as some movies might have the same title but this can evolve. I created a new [issue](https://github.com/iai-group/UserSimCRS/issues/137) to investigate this.